Fix version constraint conflict which currently brakes integration tests

```
[terraform-google-module-template]$ make test
./test/test.sh
---
driver:
  variables:
    project_id: ####
make[1]: Entering directory /####/terraform-google-module-template/staging/terraform-google-module-test'
docker run --rm -it \
        -e PROJECT_ID \
        -e SERVICE_ACCOUNT_JSON \
        -e GOOGLE_APPLICATION_CREDENTIALS=/cft/workdir/credentials.json \
        -v "/####/terraform-google-module-template/staging/terraform-google-module-test":/cft/workdir \
        gcr.io/cloud-foundation-cicd/cft/kitchen-terraform:2.0.0 \
        make test_integration
Activated service account credentials for: [###@###.iam.gserviceaccount.com]
test/ci_integration.sh
+ kitchen create ''
-----> Starting Kitchen (v1.24.0)
-----> Creating <simple-example-default>...
       Terraform v0.12.0
       
       Your version of Terraform is out of date! The latest version
       is 0.12.2. You can update by downloading from www.terraform.io/downloads.html
$$$$$$ Running command `terraform init -input=false -lock=true -lock-timeout=0s  -upgrade -force-copy -backend=true  -get=true -get-plugins=true -verify-plugins=true` in directory /cft/workdir/test/fixtures/simple_example
       Upgrading modules...
       - example in ../../../examples/simple_example
       - example.module_test in ../../..
       
       Error: Unsupported Terraform Core version
       
       Module example.module_test (from "../..") does not support Terraform version
       0.12.0. To proceed, either choose another supported Terraform version or
       update the module's version constraint. Version constraints are normally set
       for good reason, so updating the constraint may lead to other errors or
       unexpected behavior.
       
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Create failed on instance <simple-example-default>.  Please see .kitchen/logs/simple-example-default.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

+ finish
+ echo 'BEGIN: finish() trap handler'
BEGIN: finish() trap handler
+ kitchen destroy ''
-----> Starting Kitchen (v1.24.0)
-----> Destroying <simple-example-default>...
       Terraform v0.12.0
       
       Your version of Terraform is out of date! The latest version
       is 0.12.2. You can update by downloading from www.terraform.io/downloads.html
$$$$$$ Running command `terraform init -input=false -lock=true -lock-timeout=0s  -force-copy -backend=true  -get=true -get-plugins=true -verify-plugins=true` in directory /cft/workdir/test/fixtures/simple_example
       Initializing modules...
       
       Error: Unsupported Terraform Core version
       
       Module example.module_test (from "../..") does not support Terraform version
       0.12.0. To proceed, either choose another supported Terraform version or
       update the module's version constraint. Version constraints are normally set
       for good reason, so updating the constraint may lead to other errors or
       unexpected behavior.
       
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Destroy failed on instance <simple-example-default>.  Please see .kitchen/logs/simple-example-default.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

make: *** [Makefile:78: test_integration] Error 20
make[1]: *** [Makefile:142: test_integration_docker] Error 2
make[1]: Leaving directory '/####/terraform-google-module-template/staging/terraform-google-module-test'
make[1]: Entering directory '/####/terraform-google-module-template/staging/terraform-google-module-test'
docker run --rm -it \
        -e PROJECT_ID \
        -e SERVICE_ACCOUNT_JSON \
        -e GOOGLE_APPLICATION_CREDENTIALS=/cft/workdir/credentials.json \
        -v "/####/terraform-google-module-template/staging/terraform-google-module-test":/cft/workdir \
        gcr.io/cloud-foundation-cicd/cft/kitchen-terraform:2.0.0 \
        /bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen destroy"
Activated service account credentials for: [####@####.iam.gserviceaccount.com]
-----> Starting Kitchen (v1.24.0)
-----> Destroying <simple-example-default>...
       Terraform v0.12.0
       
       Your version of Terraform is out of date! The latest version
       is 0.12.2. You can update by downloading from www.terraform.io/downloads.html
$$$$$$ Running command `terraform init -input=false -lock=true -lock-timeout=0s  -force-copy -backend=true  -get=true -get-plugins=true -verify-plugins=true` in directory /cft/workdir/test/fixtures/simple_example
       Initializing modules...
       
       Error: Unsupported Terraform Core version
       
       Module example.module_test (from "../..") does not support Terraform version
       0.12.0. To proceed, either choose another supported Terraform version or
       update the module's version constraint. Version constraints are normally set
       for good reason, so updating the constraint may lead to other errors or
       unexpected behavior.
       
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Destroy failed on instance <simple-example-default>.  Please see .kitchen/logs/simple-example-default.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

make[1]: *** [Makefile:132: docker_destroy] Error 20
make[1]: Leaving directory '/####/terraform-google-module-template/staging/terraform-google-module-test'
make: *** [Makefile:3: test] Error 2
```

--- a/README.md
+++ b/README.md
@@ -42,6 +42,11 @@ The module will be generated at
 `./staging/terraform-google-module-test`. If the tests pass, the module
 will be removed; otherwise, it will be left in place for inspection.
 
+In order to execute the test following tools need to be installed:
+- `make`
+- `docker`
+- `jq`
+
 [cookiecutter]: https://cookiecutter.readthedocs.io/
 [kitchen-terraform]: https://github.com/newcontext-oss/kitchen-terraform
 [contributing]: ./terraform-google-{{cookiecutter.module_name}}/CONTRIBUTING.md

--- a/terraform-google-{{cookiecutter.module_name}}/main.tf
+++ b/terraform-google-{{cookiecutter.module_name}}/main.tf
@@ -15,7 +15,7 @@
  */
 
 terraform {
-  required_version = "~> 0.11.0"
+  required_version = "~> 0.12.0"
 }
 
 resource "google_storage_bucket" "main" {

